troi/__init__.py

```diff
@@ -222,36 +222,75 @@ class Artist(Entity):
 
     def __init__(self,
                  name=None,
-                 mbids=None,
+                 mbid=None,
+                 artist_id=None,
+                 join_phrase=None,
+                 ranking=None,
+                 musicbrainz=None,
+                 listenbrainz=None,
+                 acousticbrainz=None):
+        Entity.__init__(self, mbid=mbid, ranking=ranking, musicbrainz=musicbrainz,
+                        listenbrainz=listenbrainz, acousticbrainz=acousticbrainz)
+        self.name = name
+        self.artist_id = artist_id
+        self.join_phrase = join_phrase
+
+    def __str__(self):
+        return "<Artist('%s', [%s], %s)>" % (self.name, self.mbid, self.artist_id)
+
+
+class ArtistCredit(Entity):
+    """
+        The class that represents an artist credit.
+    """
+
+    def __init__(self,
+                 name=None,
+                 artists=None,
                  artist_credit_id=None,
                  ranking=None,
                  musicbrainz=None,
                  listenbrainz=None,
                  acousticbrainz=None):
-        Entity.__init__(self, ranking=ranking, musicbrainz=musicbrainz, listenbrainz=listenbrainz, acousticbrainz=acousticbrainz)
+        Entity.__init__(self, ranking=ranking, musicbrainz=musicbrainz,
+                        listenbrainz=listenbrainz, acousticbrainz=acousticbrainz)
         self.name = name
+        self.artists = artists
         self.artist_credit_id = artist_credit_id
-        if mbids:
-            if not isinstance(mbids, list) and not isinstance(mbids, tuple):
-                raise TypeError("Artist mbids must be a list.")
-            self.mbids = sorted(mbids)
-        else:
-            self.mbids = None
+        if artists:
```

Comment:
probably want to remove either this or the artists assignment 2 lines above or I guess assign to an empty list at the top. If you remove the assignment above, then add an else case to assign to empty list below otherwise there will be an AttributeError when artists don't exist.